### DISCUSSION TITLE
我有一个/home的路由但是访问/home/anything时页面是空白的，怎么能实现访问/home/anything时让它重定向到/home或者404

### DISCUSSION THREAD

**Discussion by hyl1374961656** [umijs/umi#11952] [Q&A]
我有一个/home的路由但是访问/home/anything时页面是空白的，怎么能实现访问/home/anything时让它重定向到/home或者404

**Comment by hyl1374961656** [comment]
export default defineConfig({
  antd: {},
  access: {},
  model: {},
  initialState: {},
  request: {},
  layout: false,
  publicPath: undefined,
  routes: [
    {
      path: '/',
      redirect: '/home',
    },
    {
      path: '/home',
      component: './Home',
    },
    {
      path: '/access',
      component: './Access',
    },
    {
      path: '/table',
      component: './Table',
    },
    { path: '/*', component: '@/pages/404' },
  ],
  npmClient: 'yarn',
});

好像publicPath是undefined的时候是正常的，有值就没法走404了

**Comment by GuanJdoJ** [comment]
routes配置：
- 跳到404 `{ path: '/home/*', component: '@/pages/your/404.tsx' }`
- 重定向 `{ path: '/home/*', redirect: '/home' }`

**Comment by fz6m** [comment]
第一种方式是直接在路由 `routes` 里配置 `redirect: '/some-path'` ，实现可参考楼上代码。

第二种方式是给它写一个自己的组件，在这个组件里使用 `<Redirect />` （从 `umi` 导入） 或者 `history.replace` 方法来跳转路由都可以。

**Reply by hyl1374961656** [reply]
我重新装了个项目，尝试发现是正常的，但是publicPath设置值了就没法跳转了

**Reply by fz6m** [reply]
`publicPath` 在开发时需要设定为默认值，仅在生产使用：

```ts
// .umirc.ts

  publicPath: process.env.NODE_ENV === 'development' ? '/' : '//domain.com/path/to/'
```

如有问题，需要给一个最小复现仓库。
